### PR TITLE
585 coin animation on and off switchable

### LINKED ISSUE #585: 🚀 [Feature] Coin animation on and off switchable
## 🚀 Feature
it would be good the coinanimation on and off switchable

### PR DESCRIPTION
**Pull request by ogerly**
## 🍰 Pullrequest
coin animation on and off switchable in user settings 

FireShot Capture 497 - Gradido Account - localhost

### Issues
- fixes #585
 
 todo: @Elweyn @Mogge   
 es müssen im store ein neuer punkt eingefügt werden "coinanimation"
 - dieser sollte wohl auch in der DB für den user exestieren

### CODE REVIEW

**Comment by Elweyn** [comment]
@ulfgebhardt, @clauspeterhuebner, 
@ogerly and me are speaking about this issue. And their would be some todos on the backend.
- [ ] Database User stores the coinanimation flag
- [ ] Get User in Apollo and LoginServer gives back the coinanimation flag of a User
- [ ] Update User in LoginServer can update the coinanimation flag of a User

**Comment by ogerly** [comment]
classe anpassen from style

**Comment by Elweyn** [comment]
@einhornimmond Assigned you to this task/PR since we need to store the flag to the database in the user table

**Comment by einhornimmond** [comment]
So, coinanimation will be stored in new settings table (key-value approach)
It will be saved from frontend and load by login into frontend

Attention: to get database migration to work, database_build volume must be deleted first and database must be builded. 

By testing it coinanimation don't seem to stop after disabling it and the locale for it is also missing.

File: backend/src/util/validate.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+function isStringBoolean(value: string): boolean {
+  const lowerValue = value.toLowerCase()
+  if (lowerValue === 'true' || lowerValue === 'false') {
+    return true
+  }
+  return false
+}
```
Comment (ulfgebhardt) [inline backend/src/util/validate.ts:7]: pretty sure this can be simplified

File: login_server/src/cpp/JSONInterface/JsonUpdateUserInfos.cpp [hunk 2]
```diff
@@ -34,8 +33,7 @@ Poco::JSON::Object* JsonUpdateUserInfos::handle(Poco::Dynamic::Var params)
 		/// not available for the given type.
 		/// Throws InvalidAccessException if Var is empty.
 		try {
-			paramJsonObject->get("email").convert(email);
-
+
```
Comment (ulfgebhardt) [inline login_server/src/cpp/JSONInterface/JsonUpdateUserInfos.cpp:36]: docu missing

**Review by ulfgebhardt — CHANGES_REQUESTED** [review]

**Review by ulfgebhardt — APPROVED** [review]

**Comment by ogerly** [comment]
sehr geil

File: backend/src/graphql/resolvers/UserResolver.ts [hunk 3]
```diff
@@ -20,6 +20,10 @@ import {
   klicktippNewsletterStateMiddleware,
 } from '../../middleware/klicktippMiddleware'
 import { CheckEmailResponse } from '../models/CheckEmailResponse'
+import { getCustomRepository } from 'typeorm'
+import { UserSettingRepository } from '../../typeorm/repository/UserSettingRepository'
+import { Setting } from '../../graphql/enum/Setting'
```
Comment (Mogge) [inline backend/src/graphql/resolvers/UserResolver.ts:25]: I suggest the name Settings

File: backend/src/graphql/enum/Setting.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+enum Setting {
```
Comment (Mogge) [inline backend/src/graphql/enum/Setting.ts:1]: Settings as well the filename

**Review by Mogge — COMMENTED** [review]
I really do not understand the rush of this PR. I question the solution in creating a hash table for the user settings. Why not simply add a column to the `user_state` table with a boolean value for the coin animation? The hack to convert a string into a boolean is so ugly ...

**Pull request merged**